chef/resource_chef_node.go

```diff
@@ -35,6 +30,17 @@ func resourceChefNode() *schema.Resource {
 					Type: schema.TypeString,
 				},
 			},
+
+			"environment": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "master",
+			},
+
+			"attributes": &schema.Schema{
+				Type:     schema.TypeMap,
```

Comment:
Is `schema.TypeMap` sufficient for expressing nested attribute maps, like in the chef provisioner? i.e., will this work:
```
attributes {
  foo {
    bar = "spam and eggs"
  }
}
```